### DISCUSSION TITLE
How to use audio tag using file tag

### DISCUSSION THREAD

**Discussion by Yeong97** [community/community#21461] [Other Feature Feedback, Questions, & Ideas]
<pre><code class="lang-auto">&lt;!DOCTYPE html&gt;
&lt;!-- Author: Eric Bidelman (ericbidelman@chromium.org) --&gt;
&lt;html&gt;
&lt;head&gt;
  &lt;meta charset="utf-8" /&gt;
  &lt;meta http-equiv="X-UA-Compatible" content="chrome=1" /&gt;
  &lt;title&gt;Web Audio API: Simple load + play&lt;/title&gt;
&lt;/head&gt;
&lt;body&gt;
  &lt;p&gt;Example of using the Web Audio API to load a sound file
  and start playing on user-click.&lt;/p&gt;
  &lt;input type="file"  name="audio-src" accept="audio/*"&gt;
  &lt;audio id="audio-ctr"&gt; &lt;/audio&gt;

  &lt;button onclick="playSound()" disabled&gt;Start&lt;/button&gt;
  &lt;button onclick="stopSound()" disabled&gt;Stop&lt;/button&gt;
  
&lt;script&gt;
var context = new AudioContext();
var source = null;
var audioBuffer = null;
var audio_ctr = document.getElementById("audio-ctr");

function stopSound() {
  if(source) {
    source.stop();
  }
}

function playSound() {
  // source is global so we can call .noteOff() later.
    
 
  
  audio_ctr.controls = true;
     
  source = context.createBufferSource();
  source.buffer = audioBuffer;
  source.loop = false;
  source.connect(context.destination);
  source.start(); // Play immediately.
}

function initSound(arrayBuffer) {
  context.decodeAudioData(
    arrayBuffer,
    function (buffer) {
     
      audioBuffer = buffer;
      var buttons = document.querySelectorAll("button");
      buttons[0].disabled = false;
      buttons[1].disabled = false;
    },
    function (e) {
      console.log("Error decoding file", e);
    }
  );
}

// User selects file, read it as an ArrayBuffer and pass to the API.
var fileInput = document.querySelector('input[type="file"]');
fileInput.addEventListener(
  "change",
  function (e) {
 
    var reader = new FileReader(); //src에 리더에 소스값 넣어보기
    reader.onload = function (e) {
     
      initSound(this.result);
     
     
    };
    reader.readAsArrayBuffer(this.files[0]);
  },
  false
);

function loadSoundFile(url) {
  var xhr = new XMLHttpRequest();
  xhr.open("GET", url, true);
  xhr.responseType = "arraybuffer";
  xhr.onload = function (e) {
    initSound(this.response); // this.response is an ArrayBuffer.
  };
  xhr.send();
}
&lt;/script&gt;
&lt;/body&gt;
&lt;/html&gt;
</code></pre>
<p>I wanted to apply the mp3 file uploaded using the file tag to the audio tag. I tried various attempts, but so far I have only been able to activate the controller of the audio tag.<br>
I am going to use the playback control box by applying the mp3 file uploaded using the file tag to the audio tag. Is there no way? Please help if there is a way.</p>

**Comment by billythekid** [comment]
<p>Hi again,</p>
<p>I think you want to load the file as an audio source when selected? The following stripped down code should do that:</p>
<p>HTML:</p>
<pre><code class="lang-auto">&lt;input type="file" name="audio-src" accept="audio/*"&gt;
&lt;audio id="audio-ctr"&gt;&lt;/audio&gt;
</code></pre>
<p>Javascript:</p>
<pre><code class="lang-javascript">var audio_ctr = document.getElementById("audio-ctr");
var fileInput = document.querySelector('input[type="file"]');

fileInput.addEventListener(
  "change",
  function () {
    var reader = new FileReader();
    reader.addEventListener("load", function () {
      audio_ctr.src = this.result;
      audio_ctr.controls = true;
    });
    reader.readAsDataURL(this.files[0]);
  },
  false
);
</code></pre>
<p>Cheers,<br>
Billy :cowboy_hat_face:</p>
